homie/src/lib.rs

```diff
@@ -311,7 +416,10 @@ impl HomieDevice {
 
     /// Remove the node with the given ID.
     pub async fn remove_node(&mut self, node_id: &str) -> Result<(), SendError<Request>> {
-        self.nodes.retain(|n| n.id != node_id);
+        // Panic on attempt to remove a node which was never added.
+        let index = self.nodes.iter().position(|n| n.id == node_id).unwrap();
```

Comment:
(optional: probably when we work out how we're *supposed* to be doing error handling): we're already returning a result. Could we add a variant for "you tried to remove a node that wasn't there" rather than panicking?